### ISSUE TITLE
Admin color palette: Sass variables / design tokens renaming

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#8883]
Follow-up to #8824, in particular [Scott’s comment](https://github.com/wagtail/wagtail/pull/8824#pullrequestreview-1037839858). Now that almost all of Wagtail’s colors are defined from our design tokens, we need to refine our approach to Sass variables, so their usage makes sense with our design tokens. And maybe review the naming of our design tokens.

## Sass variables

Particularly problematic sets of variables are the teals (secondary) and greys:

```scss
$color-teal: theme('colors.secondary.DEFAULT');
$color-teal-darker: theme('colors.secondary.400');
$color-teal-dark: theme('colors.secondary.600');

$color-grey-1: theme('colors.grey.600');
$color-grey-2: theme('colors.grey.400');
$color-grey-3: theme('colors.grey.200');
$color-grey-4: theme('colors.grey.100');
$color-grey-5: theme('colors.grey.50');
```

Those are two sets of variables with non-semantic names that differ from the design tokens’ names. Some of the options to resolve this are to:

- Rename those variables to use the same naming scheme as design tokens
- Remove those variables and use non-semantic design tokens directly
- Replace those variables with semantic ones (for example `$color-teal-dark` would be replaced by `$color-button-bicolor-hover`.
- Remove those variables, replace non-semantic design tokens with semantic ones, and use semantic design tokens directly (`theme('colors.button.bicolor.hover')`

## Design tokens

There are a couple points we need to solidify:

- Usage of numeric scales for the shade of each color – whether the current numbers make enough sense, or whether we should change them (if so how)
- How (and whether) to use "default" shades for given colors

## Additional areas to consider

As part of this we could also consider:

- The addition of more design tokens for the remaining colors that aren’t customisable: addition and deletion highlights, and the focus outline.
- Changes to other Sass variables / design tokens than the above two sets.
- How we use fades/transparencies of white/black
- How we use shadows

Ideally I think we should also think of how those types of setups can help support the introduction of a dark theme, and high-contrast light/dark themes (WCAG 2.1 AAA 7:1 contrast or above).

**Comment by thibaudcolas** [comment]
I’d like to work on this as part of #10056, making sure we end up with one set of variables that supports multiple themes.

**Comment by thibaudcolas** [comment]
Implemented as part of #10056. I went for: _Replace those variables with semantic ones_.

There is room for improvement on the set of semantic icons – we’ve defined them based on _future_ designs of Wagtail with consolidated colors, rather than the as-is CMS with more variations. So there was some consolidation already but potentially too much.
